public/javascripts/SVLabel/src/SVLabel/canvas/ContextMenu.js

```diff
@@ -472,15 +472,38 @@ function ContextMenu (uiContextMenu) {
             _setTags(targetLabel);
             _setTagColor(targetLabel);
             if (getStatus('disableTagging')) { disableTagging(); }
+
+            // Hide the severity menu for the Pedestrian Signal label type.
+            if (labelType === 'Signal') {
+                $severityMenu.css({visibility: 'hidden', height: '0px'});
+            } else {
+                $severityMenu.css({visibility: 'inherit', height: '50px'});
+            }
             var windowHeight = $menuWindow.outerHeight();
+            var connectorBuffer = 3; // for connector to overlap border of label icon
+
+            var connectorHeight = parseInt(window.getComputedStyle($connector[0]).getPropertyValue("height"));
+            var connectorWidth = parseInt(window.getComputedStyle($connector[0]).getPropertyValue("width"));
 
             // Determine coordinates for context menu when displayed below the label.
-            var topCoordinate = labelCoord.y + 20;
-            var connectorCoordinate = -5;
+            var topCoordinate = labelCoord.y
+                + svl.LABEL_ICON_RADIUS
+                + parseInt(connectorHeight)
+                - connectorBuffer;
+            var connectorCoordinate = -1 * parseInt(connectorHeight);
 
             // Determine coordinates for context menu when displayed above the label.
-            if (labelCoord.y + windowHeight + 22 > util.EXPLORE_CANVAS_HEIGHT) {
-                topCoordinate = labelCoord.y - windowHeight - 22;
+            // labelCoord is top-left of label but should be center of rendered label, so must account for icon radius
```

Comment:
Reminder to check the [Style Guide](https://github.com/ProjectSidewalk/SidewalkWebpage/wiki/Style-Guide) for comment style! Though if the first work is the name of a variable, you don't need to capitalize ;)